client/src/core/ErrorBoundary.js

```diff
@@ -1,126 +1,230 @@
+        error: null,
+        testing_for_stale_client: false,
+      };
+    }
+    static getDerivedStateFromError(error) {
+      return {
+        error: error,
+        testing_for_stale_client: true,
+      };
+    }
+    catch_stale_client_error_case() {
+      const unique_query_param =
+        Date.now() + Math.random().toString().replace(".", "");
 
-    // Stale clients are our most likely production errors, always check for and attempt to handle them
-    // That is, reload the page without cache if the client/CDN sha's are mismatched (and the build is non-dev)
-    // Otherwise, log the error (again, if non-dev) and display error component
-    make_request(get_static_url("build_sha", unique_query_param))
-      .then((build_sha) => {
-        const local_sha_matches_remote_sha =
-          build_sha.search(`^${window.sha}`) !== -1;
+      // Stale clients are our most likely production errors, always check for and attempt to handle them
+      // That is, reload the page without cache if the client/CDN sha's are mismatched (and the build is non-dev)
+      // Otherwise, log the error (again, if non-dev) and display error component
+      make_request(get_static_url("build_sha", unique_query_param))
+        .then((build_sha) => {
+          const local_sha_matches_remote_sha =
+            build_sha.search(`^${window.sha}`) !== -1;
 
-        if (!local_sha_matches_remote_sha && !window.is_dev) {
-          window.location.reload(true);
-        } else {
+          if (!local_sha_matches_remote_sha && !window.is_dev) {
+            window.location.reload(true);
+          } else {
+            this.log_error_and_display_error_page();
+          }
+        })
+        .catch(() => {
           this.log_error_and_display_error_page();
-        }
-      })
-      .catch(() => {
-        this.log_error_and_display_error_page();
-      });
-  }
-  log_error_and_display_error_page() {
-    if (!window.is_dev) {
-      log_standard_event({
-        SUBAPP: window.location.hash.replace("#", ""),
-        MISC1: "ERROR_IN_PROD",
-        MISC2: this.state.error.toString(),
+        });
+    }
+    log_error_and_display_error_page() {
+      if (!window.is_dev) {
+        log_standard_event({
+          SUBAPP: window.location.hash.replace("#", ""),
+          MISC1: "ERROR_IN_PROD",
+          MISC2: this.state.error.toString(),
+        });
+      }
+
+      this.setState({
+        testing_for_stale_client: false,
       });
+
+      throw this.state.error;
     }
+    render() {
+      const { children, set_redirect_msg, location } = this.props;
+      const { error, testing_for_stale_client } = this.state;
 
-    this.setState({
-      testing_for_stale_client: false,
-    });
+      if (_.isNull(error)) {
+        return children;
+      }
+      if (testing_for_stale_client) {
+        this.catch_stale_client_error_case();
+        return null;
+      } else {
+        const redirect_to_parent_dept_or_home = (level, subj_id) => {
```

Comment:
I think all the infographic redirect stuff should live in the infographic route rather than the `ErrorBoundary`. Would prefer to only try and handle generic/app-wide errors at this level.

Reply:
If we want sub-app specific redirect rules in other places, maybe it could be a configurable part of the `StandardRouteContainer` or it's own higher order component for wrapping routes in.